flake.nix

```
@@ -0,0 +1,47 @@
+{
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs/nixpkgs-unstable";
+    utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs =
+    {
+      self,
+      nixpkgs,
+      utils,
+    }:
+    utils.lib.eachDefaultSystem (
+      system:
+      let
+        pkgs = nixpkgs.legacyPackages.${system};
+      in
+      {
+        formatter = pkgs.nixfmt-tree;
+        packages = rec {
+          default = tarts;
+          tarts =
+            let
+              manifest = pkgs.lib.importTOML ./Cargo.toml;
+            in
+            pkgs.rustPlatform.buildRustPackage {
+              pname = manifest.package.name;
+              version = manifest.package.version;
+
+              src = pkgs.lib.cleanSource ./.;
+              cargoLock.lockFile = ./Cargo.lock;
```

forgot to mention earlier but adding mainProgram will allow for using `nix run .#tarts -- my-args`
```suggestion
              cargoLock.lockFile = ./Cargo.lock;
              meta.mainProgram = "tarts";
```